test/test.js

```diff
@@ -347,4 +396,319 @@ describe('rollup-plugin-babel', function() {
 			assert.ok(code.indexOf('console.foobaz') !== -1);
 		});
 	});
+
+	it('can be used as an input plugin while transforming the output', () => {
+		return rollup
+			.rollup({
+				input: 'samples/basic/main.js',
+				plugins: [
+					babelPlugin.generated({
+						presets: ['@babel/env'],
+					}),
+				],
+			})
+			.then(bundle => bundle.generate({ format: 'cjs' }))
+			.then(({ output: [{ code }] }) => {
+				assert.ok(code.indexOf('const') === -1, code);
+			});
+	});
+});
+
+describe('rollup-plugin-babel when used as an output plugin', function() {
+	this.timeout(15000);
+
+	function bundleWithOutputPlugin(input, babelOptions = {}, generateOptions = {}, rollupOptions = {}) {
+		return rollup
+			.rollup(Object.assign({ input }, rollupOptions))
+			.then(bundle =>
+				bundle.generate(
+					Object.assign(
+						{
+							format: 'cjs',
+							plugins: [babelPlugin.generated(babelOptions)],
+						},
+						generateOptions,
+					),
+				),
+			)
+			.then(({ output: [generated] }) => generated);
+	}
+
+	it('allows running the plugin on the output via output options', () => {
+		return bundleWithOutputPlugin('samples/basic/main.js', {
+			presets: ['@babel/env'],
+		}).then(({ code }) => {
+			assert.ok(code.indexOf('const') === -1, code);
+		});
+	});
+
+	it('ignores .babelrc when transforming the output by default', () => {
+		return bundleWithOutputPlugin('samples/basic/main.js').then(({ code }) => {
+			assert.ok(code.indexOf('const') !== -1, code);
+		});
+	});
+
+	it('allows transform-runtime to be used instead of bundled helpers', () => {
+		return bundleWithOutputPlugin(
+			'samples/runtime-helpers/main.js',
+			{
+				presets: [['@babel/env', { modules: 'cjs' }]],
+				plugins: ['@babel/transform-runtime'],
+			},
+			{ format: 'esm' },
```

Comment:
I think the implementation of this test is somewhat too specific for what it is supposed to test. A combination of `{ modules: 'cjs' }` and `{ format: 'esm' }` looks special - but it has nothing to do with a simple thing of "allows transform-runtime to be used instead of bundled helpers".

I think this could have been written like this because of the problem we've discussed here - https://github.com/rollup/rollup-plugin-babel/pull/333/files#r348753235 , if yes - then it can be updated now as the issue got fixed.

Reply:
Updated